src/diffusers/training_utils.py

```diff
@@ -39,88 +40,173 @@ def set_seed(seed: int):
     # ^^ safe to call this function even if cuda is not available
 
 
+# Adapted from torch-ema https://github.com/fadel/pytorch_ema/blob/master/torch_ema/ema.py#L14
 class EMAModel:
     """
     Exponential Moving Average of models weights
     """
 
     def __init__(
         self,
-        model,
-        update_after_step=0,
-        inv_gamma=1.0,
-        power=2 / 3,
-        min_value=0.0,
-        max_value=0.9999,
-        device=None,
+        parameters: Iterable[torch.nn.Parameter],
+        decay: float = 0.9999,
+        update_after_step: int = 0,
+        use_ema_warmup: bool = False,
+        inv_gamma: Union[float, int] = 1.0,
+        power: Union[float, int] = 2 / 3,
```

Comment:
supported both ema schedules here; the only change is I removed the `min_value` and `max_value` and just added `decay`.